### PR TITLE
feat: declare filters with #[DataTableFilter]

### PR DESCRIPTION
**Pull request by pentiminax**
Step 4 of the attribute harmonization tracked in #470, on top of #474. This closes the coverage gap the design opened with: columns were annotatable, the six filter classes were not, and nothing explained the boundary.

## What changes

`#[DataTableFilter]` carries the same four arguments as `#[DataTableColumn]`, in the same order, and resolves its options through the same three-step convention.

```php
use Pentiminax\UX\DataTables\Attribute\DataTableFilter;

class User
{
    #[DataTableFilter(options: ['label' => 'Email'])]
    private string $email;

    #[DataTableFilter]
    private Status $status;

    #[DataTableFilter(options: ['label' => 'Created'])]
    private \DateTimeImmutable $createdAt;
}
```

No `configureFilters()` needed. The filter class is guessed from the declared type:

| Declared type | Filter |
| --- | --- |
| `bool` | `TernaryFilter` |
| A `BackedEnum` | `ChoiceFilter`, options filled from the cases |
| A `DateTimeInterface` | `DateRangeFilter` |
| Anything else, or no type | `TextFilter` |

A `ChoiceFilter` guessed from an enum fills its own options, since leaving them empty would render an empty select. An explicit `options` entry still wins.

## CheckboxFilter is rejected, not guessed

`CheckboxFilter` has no default condition — it exists to run a `query()` closure, and an attribute cannot carry a closure. One built from an attribute would throw on the first request that submits it, which is the silent failure mode these steps exist to remove. Declaring it now fails the container build with a message saying why.

Rejected alternative: a `requiresQueryCallback()` method on the filter contract. That is a public API change to express one class's property.

## Precedence

Filters resolve down the same chain as columns, first level that declares anything winning:

1. `configureFilters()`
2. `#[DataTableFilter]` on the table class
3. `#[DataTableFilter]` on the data class

`ValidateDataTableAttributesPass` validates them with the same short-circuit it gained in #474, so shadowed declarations do not fail the build.

## Backward compatibility

Nothing existing can reach the new paths: the attribute did not exist, and a table whose `configureFilters()` declared nothing kept an empty collection for good. `DataTableInfrastructure::createDefault()` gains an optional last parameter.

The impact run on `initialize()` and `createDefault()` comes back CRITICAL, because both sit at the centre of every render. Both edits are additive in the sense above.

## Verification

- `vendor/bin/phpunit` — 1753 tests, 6326 assertions, OK
- `composer fix`, `composer audit` — clean
- `npm run lint`, `npm run build` in `docs/` — clean
- Sabotage runs: dropping the checkbox rejection fails 2 tests, dropping the enum option filling fails 1, removing the `initialize()` block fails 2

Not tested: rendering a filter bar built from attributes end to end in a browser. Coverage stops at the serialized `Filters` collection.

## Documentation

`reference/filters.mdx` gains the attribute section, `guide/filters.mdx` the short form plus a caution that `configureFilters()` wins wholesale rather than merging, `reference/attributes.mdx` the cross-reference. Examples still passing `entityClass:` were moved to `dataClass:`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/DependencyInjection/Compiler/ValidateDataTableAttributesPass.php [hunk 1]
```diff
@@ -37,24 +40,53 @@ public function process(ContainerBuilder $container): void
 
             $asDataTable = $this->guard($class, $class, static fn () => $resolver->resolve($class));
 
-            $classColumns = $this->guard($class, $class, static fn () => $reader->readClassColumns($class));
+            $readsColumnAttributes = !$this->overrides($class, 'configureColumns');
+            $readsFilterAttributes = !$this->overrides($class, 'configureFilters');
```
[the remark this answers is not in the document]
Comment (pentiminax) [reply src/DependencyInjection/Compiler/ValidateDataTableAttributesPass.php:44]: Not a defect: this is the documented trade-off of b3b1a227, and it is in direct tension with the earlier finding on this same file.

The pass cannot know what `configureFilters()` returns without instantiating the table, which container compilation will not do. That leaves two options:

- validate regardless, and fail `cache:clear` over declarations a table with its own `configureFilters()` never reads — the previous finding, which was accepted;
- skip the override, and let the conditional-empty case surface on the first request, where it surfaced before the check existed at all.

The second is chosen deliberately: a broken build blocks an entire application, including parts unrelated to the table. It is stated in the commit message and in both Validation sections of the docs.

The fixtures were fair criticism though. `OwnFiltersTableFixture` returned an empty collection, which illustrated the edge case rather than the path the test is about. Both fixtures now declare a filter and a column the way a real table does (3939d551).
